### DISCUSSION TITLE
Trigger when a stored procedure is called

### DISCUSSION THREAD

**Discussion by praj18** [PostgREST/postgrest#2056] [Q&A] [answered]
As far as I know, triggers are fired only when there are changes to a table, however, I was wondering if there was a way to fire a trigger when another procedure is called.

I am trying to implement page visit counts for each webpage that I have, however, I do not need the visit count immediately as soon as I visit the page, hence, I didn't opt to just run an incremental query whenever the page is loaded. 

I instead want to run a trigger that adds to the page visit count whenever I run a stored procedure of retrieving the webpage data. 

Is this in any way possible? If not, are there any better solutions?

**Comment by steve-chavez** [comment]


**Reply by praj18** [reply]
Thank you for your reply. However, how would I know that the page has been visited? Could I run a cronjob based on a stored procedure?

**Reply by steve-chavez** [reply]
I was thinking you could schedule the cron to run sometime after the visit happens, so it would be a `cron.schedule` call at the end of your stored procedure which updates the visit count.

**Reply by steve-chavez** [reply]
@praj18 Have a look at the other answer, I made the wrong suggestion thinking this was another org/repo :facepalm:

**Comment by wolfgangwalther** [comment, marked as answer]
Did you look into the pre-request hook that PostgREST provides?

Maybe that could help: https://postgrest.org/en/v9.0/configuration.html#pre-request
